imap/jmap_mailbox.c

```diff
@@ -2920,6 +2922,11 @@ static void _mbox_destroy(jmap_req_t *req, const char *mboxid,
         goto done;
     }
 
+    /* Do not allow to remove $scheduled */
+    if (mboxname_isscheduledmailbox(mbentry->name, mbentry->mbtype)) {
```

Comment:
We shouldn't allow deletion of specialuse folders at all, so this should be subsumed by that rather than an explicit check for the scheduled mailbox.